wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift

```diff
@@ -620,29 +620,30 @@ extension ConversationViewController: ConversationInputBarViewControllerDelegate
     }
 
     var searchBarButtonItem: UIBarButtonItem {
-        let showingSearchResults = (self.collectionController?.isShowingSearchResults ?? false)
-        let action = #selector(ConversationViewController.onCollectionButtonPressed(_:))
+        let button = UIButton(type: .system)
+        button.setImage(UIImage(resource: .search), for: .normal)
+        button.tintColor = IconColors.foregroundDefault
 
-        let button = IconButton()
-        button.setIcon(showingSearchResults ? .activeSearch : .search, size: .tiny, for: .normal)
         button.accessibilityIdentifier = "collection"
         button.accessibilityLabel = L10n.Accessibility.Conversation.SearchButton.description
 
-        button.addTarget(self, action: action, for: .touchUpInside)
+        button.addTarget(self,
+                        action: #selector(onSearchButtonPressed(_:)),
+                        for: .touchUpInside)
```

Comment:
I couldn’t migrate to the `UIAction` pattern here because `onSearchButtonPressed` is also being used in the initialization of a `UIKeyCommand`, which only accepts a `Selector`. `UIKeyCommand` doesn’t have an initializer that supports a `UIAction`, hence the use of the `#selector` approach.